flashinfer/quantization/mxfp8_layout.py

```
@@ -0,0 +1,309 @@
+        INT32-packed UE8M0 scale tensor in per-row TMA-aligned MN-major layout.
+        Shape ``(m, ceil(k / (k_gran * 4)))`` for 2D input or
+        ``(b, m, ceil(k / (k_gran * 4)))`` for 3D input, dtype ``torch.int32``.
+    """
+    assert input.dtype == torch.bfloat16, f"input must be bfloat16; got {input.dtype}"
+    assert k_gran in (32, 128), f"k_gran must be 32 or 128; got {k_gran}"
+    assert input.size(-1) % k_gran == 0, (
+        f"input.size(-1)={input.size(-1)} not divisible by k_gran={k_gran}"
+    )
+    if input.numel() == 0:
+        import math
+        return torch.empty_like(input, dtype=torch.float8_e4m3fn), torch.empty(input.shape[:-1] + (math.ceil(input.shape[-1] / (k_gran * 4)),), dtype=torch.int32, device=input.device)
```

Ruff (0.15.15)

[warning] Consider iterable unpacking instead of concatenation

Replace with iterable unpacking

(RUF005)